src/sql/workbench/contrib/notebook/browser/cellViews/NotebookFindDecorations.ts

```diff
@@ -0,0 +1,330 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { IDisposable } from 'vs/base/common/lifecycle';
+import { IModelDecorationsChangeAccessor, IModelDeltaDecoration, OverviewRulerLane, TrackedRangeStickiness, MinimapPosition } from 'vs/editor/common/model';
+import { ModelDecorationOptions } from 'vs/editor/common/model/textModel';
+import { overviewRulerFindMatchForeground, minimapFindMatch } from 'vs/platform/theme/common/colorRegistry';
+import { themeColorFromId } from 'vs/platform/theme/common/themeService';
+import { NotebookEditor } from 'sql/workbench/contrib/notebook/browser/notebookEditor';
+import { NotebookRange, NotebookFindMatch } from 'sql/workbench/contrib/notebook/browser/models/modelInterfaces';
+
+export class FindDecorations implements IDisposable {
```

Comment:
can you help remind me if this code was adapted from another file? itll help me understand what changes were made that are notebook specific versus generic changes

Reply:
Yes, we copied this from vs/editor/contrib/find/findDecorations.ts

Reply:
Did you make changes to this file after copying?

---
In reply to: [349752319](https://github.com/microsoft/azuredatastudio/pull/8426#discussion_r349752319) [](ancestors = 349752319)

Reply:
Yes! findDecirations from vscode is specific to applying decorations on a code-editor. We updated it to accommodate notebook specifics in this file.

Reply:
So my worry is that in the future vs code makes changes to their FindDecorations implementation that the editor expects and thus causes us problems. Let's sit down and go over this together so we can make sure that isn't going to be a problem.

We might want to look into extending/implementing the base vs class to help with that if so as  a precaution